### PR TITLE
Expose Index and Range with adding tests

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

File: src/System.Memory/tests/Span/Indexer.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            Assert.Equal('H', span[new Index(0, fromEnd: false)]);
+            Assert.Equal('o', span[new Index(1, fromEnd: true)]);
+            Assert.Equal(span[new Index(2, fromEnd: false)], span[new Index(span.Length - 2, fromEnd: true)]);
+
+            Assert.Throws<IndexOutOfRangeException>(() =>
+            {
+                char c;
+                ReadOnlySpan<char> s = "Hello".AsSpan();
+                c = s[new Index(0, fromEnd: true)];
+                return;
```
Comment (stephentoub) [inline src/System.Memory/tests/Span/Indexer.cs]: These ending `return;` statements are not necessary.

File: src/System.Memory/tests/Span/Indexer.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            });
```
Comment (stephentoub) [inline src/System.Memory/tests/Span/Indexer.cs]: You could make this whole thing a one-liner with:
```C#
Assert.Throws<IndexOutOfRangeException>(() => "Hello".AsSpan()[new Index(0, fromEnd: true)]);
```
Same for other similar cases.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by tarekgh** [comment]
I am looking at the failure in UWP. properly I'll need to exclude the new APIs from the App compat checks.

**Comment by tarekgh** [comment]
The failed CI case is tracked by https://github.com/dotnet/corefx/issues/33389

**Pull request merged**